src/util/authtok.c

```diff
@@ -792,6 +792,78 @@ static errno_t sss_authtok_set_passkey_from_blob(struct sss_auth_token *tok,
     return ret;
 }
 
+errno_t sss_authtok_set_passkey_pin(struct sss_auth_token *tok,
+                                    const char *pin)
+{
+    TALLOC_CTX *tmp_ctx;
+    char *prompt = NULL;
+    char *key = NULL;
+    char *old_pin = NULL;
+    int ret;
+
+    if (!tok) {
+        return EINVAL;
+    }
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_OP_FAILURE, "talloc_new failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    switch (tok->type) {
+    case SSS_AUTHTOK_TYPE_EMPTY:
+        ret = ENOENT;
+        break;
+    case SSS_AUTHTOK_TYPE_PASSKEY:
+        sss_authtok_set_empty(tok);
```

Comment:
same as below.

Reply:
You are right, I've added a debug message